src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1762 @@
+        }
+
+        /// <summary>
+        /// List all targets cards in range
+        /// When using this function, you can't determine which target you've read
+        /// So you'll need to use the Decode functions to try to get a card type
+        /// So use this function only with a specific card type. Prefer the AutoPoll function
+        /// As the type identified is returned
+        /// </summary>
+        /// <param name="maxTarget">The maximum number of targets</param>
+        /// <param name="targetBaudRate">The baud rate to use to find cards</param>
+        /// <param name="initiatorData">Specific initialization data</param>
+        /// <returns>A raw byte array with the data of the targets if any has been identified</returns>
+        public byte[] ListPassiveTarget(MaxTarget maxTarget, TargetBaudRate targetBaudRate, ReadOnlySpan<byte> initiatorData)
+        {
+            Span<byte> toSend = stackalloc byte[2 + initiatorData.Length];
+            toSend[0] = (byte)maxTarget;
+            toSend[1] = (byte)targetBaudRate;
+            if (initiatorData.Length > 0)
+                initiatorData.CopyTo(toSend.Slice(2));
+            var ret = WriteCommand(CommandSet.InListPassiveTarget, toSend);
+            if (ret < 0) return null;
+            // TODO: check what is the real maximum size
+            Span<byte> listData = stackalloc byte[1024];
+            ret = ReadResponse(CommandSet.InListPassiveTarget, listData);
+            LogInfo.Log($"{nameof(ListPassiveTarget)}: {ret}, number tags: {listData[0]}", LogLevel.Debug);
+            if ((ret >= 0) && (listData[0] > 0))
+                return listData.Slice(0, ret).ToArray();
+            return null;
+        }
+
+        /// <summary>
+        /// Try to decode a raw byte array containing target information
+        /// to a 106 kbps Type A card
+        /// </summary>
+        /// <param name="toDecode">The raw byte array</param>
+        /// <returns>A decoded card of null if it can't</returns>
+        public Data106kbpsTypeA TryDecode106kbpsTypeA(Span<byte> toDecode)
+        {
+            try
+            {
+                Data106kbpsTypeA data = new Data106kbpsTypeA();
+                data.TargetNumber = toDecode[0];
+                data.Atqa = BinaryPrimitives.ReadUInt16BigEndian(toDecode.Slice(1));
+                data.Sak = toDecode[3];
+                data.NfcId = new byte[toDecode[4]];
+                for (int i = 0; i < data.NfcId.Length; i++)
+                    data.NfcId[i] = toDecode[5 + i];
+                if ((5 + data.NfcId.Length) > toDecode.Length)
+                {
+                    data.Ats = new byte[toDecode[5 + data.NfcId.Length]];
+                    for (int i = 0; i < data.Ats.Length; i++)
+                        data.Ats[i] = toDecode[6 + data.NfcId.Length + i];
+                }
+                return data;
+            }
+            catch (ArgumentOutOfRangeException)
```

Comment:
if there is an easy fix to make this and below not catch exception to detect errors that would be preferred

Reply:
No easy fix, with the function returning the array as it does, there are not that many other options. I would have prefer to have an easy way.